### PR TITLE
Feature/2786/publish tool tips

### PR DESCRIPTION
**Pull request by ByteMap**
Displayed tooltip for disabled publish button

dockstore/dockstore#2786

Displayed a tooltip detailing why the Publish button is grayed out for
any tools or workflows that does not have a valid version.

Wrapped the publish button element in a div that contains the matToolTip
as this is a workaround to the fact that disabled elements do not fire
events

### CODE REVIEW

File: src/app/workflow/version-modal/version-modal.component.html [hunk 1]
```diff
@@ -54,7 +54,14 @@ <h4 mat-dialog-title>{{ isPublic || !canWrite ? 'View' : 'Edit' }} {{ (isService
               required
               [matTooltip]="tooltip.workflowPath"
               placeholder="e.g. CancerCollaboratory/dockstore-tool-liftover"
-              [readonly]="(isService$ | async) || isPublic || !canWrite || workflow?.mode === WorkflowType.ModeEnum.HOSTED"
+              [readonly]="
+                (isService$ | async) ||
+                isPublic ||
+                !canWrite ||
+                workflow?.mode === WorkflowType.ModeEnum.HOSTED ||
+                version.frozen ||
```
Comment (coverbeck) [inline src/app/workflow/version-modal/version-modal.component.html]: Did you change this code? It looks like more than just a formatting change.

File: src/app/workflow/version-modal/version-modal.component.html [hunk 2]
```diff
@@ -104,7 +111,7 @@ <h4 mat-dialog-title>{{ isPublic || !canWrite ? 'View' : 'Edit' }} {{ (isService
           <div
             class="col-sm-9 col-md-9 col-lg-9"
             [ngClass]="{ 'col-sm-offset-3': testParameterFilePaths.length > 0 }"
-            *ngIf="!isPublic && !(isService$ | async)"
+            *ngIf="!isPublic && !(isService$ | async) && !version.frozen"
```
Comment (coverbeck) [inline src/app/workflow/version-modal/version-modal.component.html]: Same question

**Comment by garyluu** [comment]
I think there's some issues when switching base branch.

File: src/app/shared/entry-actions/entry-actions.service.spec.ts [hunk 3]
```diff
@@ -96,4 +107,18 @@ describe('Service: EntryActionsService', () => {
     expect(service.getViewPublicButtonTooltip(EntryType.BioWorkflow)).toBe('Go to the public page for this workflow');
     expect(service.getViewPublicButtonTooltip(EntryType.Service)).toBe('Go to the public page for this service');
   }));
+  it('should display message for disabled tooltip', inject([EntryActionsService], (service: EntryActionsService) => {
+    const entry = <Entry>{};
+    entry.workflowVersions = [];
```
[the remark this answers is not in the document]
Comment (garyluu) [reply src/app/shared/entry-actions/entry-actions.service.spec.ts]: `entry.workflowVersions = null` i meant

**Pull request merged**